pkg/server/image_pull.go

```diff
@@ -120,6 +121,10 @@ func (c *criService) PullImage(ctx context.Context, r *runtime.PullImageRequest)
 	}
 
 	pullOpts = append(pullOpts, c.encryptedImagesPullOpts()...)
+	if c.config.ContainerdConfig.StargzSnapshots {
+		pullOpts = append(pullOpts,
+			containerd.WithImageHandlerWrapper(stargz.AppendInfoHandlerWrapper(ref)))
```

Comment:
`AppendInfoHandlerWrapper` is to pass image ref/digest into snapshotter. It looks for more generic and independent handler in cri-containerd. 

IMO, we can define more generics labels which can be passed into snapshotter, not just stargz :)